weblate/checks/tests/test_same_checks.py

```diff
@@ -217,6 +217,5 @@ def test_same_uppercase(self):
         self.do_test(True, ("Who is RMS?", "Who is RMS?", ""))
 
     def test_same_placeholders(self):
-        self.do_test(False, ("string $URL$", "string $URL$", "placeholders:$URL$"))
-        self.do_test(False, ("string $URL$", "string ", "placeholders:$URL$"))
-        self.do_test(True, ("string $URL$", "string $URL$", "placeholders:"))
+        self.do_test(False, ("DOMAIN", "DOMAIN", "placeholders:DOMAIN"))
+        self.do_test(True, ("PLACHOLDER", "PLACEHOLDER", "placeholders:DOMAIN"))
```

Comment:
I still think this doesn't actually test the desired behavior (what should be purpose of the tests).

1. The first test does not fire regardless placeholders settings, because upper case strings are removed later:
https://github.com/WeblateOrg/weblate/blob/91be77f33b0e872bbc91afc5bdb89b9504a688a2/weblate/checks/same.py#L175-L176
2. The second test never fires for the same reason, but should not fire anyway as the string is different.

The test should cover the behavior that it works as intended. Basically it should cover:

1. The check does fire on a string when no placeholder is configured.
2. The check does not fire once placeholder is set.

I'd start with something like this (not tested):

```python
self.do_test(True, ("@location@", "@location@", ""))
self.do_test(False, ("@location@", "@location@", "placeholders:@location@"))
```

Reply:
Thank you @nijel, i followed your suggestion and now it should work.